spec/acceptance/apache_ssl_spec.rb

```diff
@@ -91,4 +92,61 @@ class { 'apache':
     end
   end
 
+  describe 'vhost ssl ssl_ca only' do
+    it 'runs without error' do
+      pp = <<-EOS
+        class { 'apache':
+          service_ensure       => stopped,
+        }
+
+        apache::vhost { 'test_ssl_ca_only':
+          docroot              => '/tmp/test',
+          ssl                  => true,
+          ssl_cert             => '/tmp/ssl_cert',
+          ssl_key              => '/tmp/ssl_key',
+          ssl_ca               => '/tmp/ssl_ca',
+          ssl_verify_client    => 'test',
+        }
+      EOS
+      apply_manifest(pp, :catch_failures => true)
```

Comment:
in our acceptance tests, we use a pattern to ensure idempotency that applies the manifest twice; the second time ensures nothing has changed.
```puppet
apply_manifest(pp, :catch_failures => true)
apply_manifest(pp, :catch_changes => true)
```

Reply:
There's even a ```shared_examples "a idempotent resource"``` for this in the spec helper so you can simply use ```it_behaves_like "a idempotent resource"``` here instead.

Reply:
Added the idempotency tests to all 4 tests modified.